scripts/cluster/distributed_op.py

```diff
@@ -15,90 +15,123 @@
-                remote_op["callback"] = token.rstrip()
-                # TODO: handle ssl verification
-                res = requests.post(
-                    "https://{}/{}/configure".format(node_ep, CLUSTER_API),
-                    json=remote_op,
-                    verify=False,
-                )
-                if res.status_code != 200:
-                    print(
-                        "Failed to perform a {} on node {} {}".format(
-                            remote_op["action_str"], node_ep, res.status_code
-                        )
-                    )
-        except subprocess.CalledProcessError as e:
-            print("Could not query for nodes")
-            raise SystemExit(e)
-        except requests.exceptions.RequestException as e:
-            print("Failed to reach node.")
-            raise SystemExit(e)
+        hostname = socket.gethostname()
+        with open(callback_token_file, "r+") as fp:
+            token = fp.read()
+
+        subprocess.check_call(
+            [MICROK8S_STATUS, "--wait-ready", "--timeout=60"],
+            stdout=subprocess.DEVNULL,
+            stderr=subprocess.DEVNULL,
+        )
+        stdout = subprocess.check_output([KUBECTL, "get", "node", "-o", "json"])
+        info = json.loads(stdout)
+        for node_info in info["items"]:
+            node_ip = get_internal_ip_from_get_node(node_info)
+            if not include_self and is_same_server(hostname, node_ip):
+                continue
+
+            nodes.append(("{}:25000".format(node_ip), token.rstrip()))
     else:
         with open(callback_tokens_file, "r+") as fp:
             for _, line in enumerate(fp):
-                parts = line.split()
-                node_ep = parts[0]
+                node_ep, token = line.split()
                 host = node_ep.split(":")[0]
-                print("Applying to node {}.".format(host))
+
                 try:
-                    # Make sure this node exists
-                    subprocess.check_call(
-                        "{}/microk8s-kubectl.wrapper get no {}".format(snap_path, host).split(),
-                        stdout=subprocess.DEVNULL,
-                        stderr=subprocess.DEVNULL,
-                    )
-                    token = parts[1]
-                    remote_op["callback"] = token
-                    # TODO: handle ssl verification
-                    res = requests.post(
-                        "https://{}/{}/configure".format(node_ep, CLUSTER_API),
-                        json=remote_op,
-                        verify=False,
-                    )
-                    if res.status_code != 200:
-                        print(
-                            "Failed to perform a {} on node {}".format(
-                                remote_op["action_str"], node_ep
-                            )
-                        )
+                    subprocess.check_call([KUBECTL, "get", "node", host])
```

Comment:
We probably need to send stdout and stderr in devnul here or else we get an output like this one:
```
$ sudo microk8s.enable dns
Infer repository core for addon dns
Enabling DNS
Applying manifest
serviceaccount/coredns created
configmap/coredns created
deployment.apps/coredns created
service/kube-dns created
clusterrole.rbac.authorization.k8s.io/coredns created
clusterrolebinding.rbac.authorization.k8s.io/coredns created
Restarting kubelet
Adding argument --cluster-domain to nodes.
NAME   STATUS   ROLES    AGE    VERSION
u2     Ready    <none>   5m1s   v1.24.1-2+d1d054585eedd9
NAME   STATUS   ROLES    AGE   VERSION
u3     Ready    <none>   39s   v1.24.1-2+d1d054585eedd9
Adding argument --cluster-dns to nodes.
NAME   STATUS   ROLES    AGE    VERSION
u2     Ready    <none>   5m2s   v1.24.1-2+d1d054585eedd9
NAME   STATUS   ROLES    AGE   VERSION
u3     Ready    <none>   40s   v1.24.1-2+d1d054585eedd9
Restarting nodes.
NAME   STATUS   ROLES    AGE    VERSION
u2     Ready    <none>   5m5s   v1.24.1-2+d1d054585eedd9
NAME   STATUS   ROLES    AGE   VERSION
u3     Ready    <none>   43s   v1.24.1-2+d1d054585eedd9
DNS is enabled

```

Reply:
good catch